Engine/DataFeeds/BaseDataExchange.cs

```diff
@@ -137,6 +137,24 @@ public void SetDataHandler(Symbol symbol, Action<BaseData> handler)
         }
 
         /// <summary>
+        /// Adds the specified hander function to handle data for the handler's symbol 
+        /// </summary>
+        /// <param name="symbol">The symbol whose data is to be handled</param>
+        /// <param name="handler">The handler to use when this symbol's data is encountered</param>
+        /// <returns>An identifier that can be used to remove this handler</returns>
+        public void AddDataHandler(Symbol symbol, Action<BaseData> handler)
+        {
+            if (_dataHandlers.ContainsKey(symbol))
+            {
+                _dataHandlers[symbol].DataEmitted += (sender, args) => handler(args); 
```

Comment:
Since `_dataHandlers` is a `ConcurrentDictionary` we should try to use those specific methods. I don't think this will cause any issues, but in theory, just because it contains the key on line 147 doesn't mean it's still there on line 149. It would be safer (and more performant) to use `_dataHandlers.TryGetValue(symbol, out dataHandler)`